esp-hal/src/i2c/master/mod.rs

```diff
@@ -2015,7 +2015,16 @@ impl Driver<'_> {
             )));
         }
         if r.arbitration_lost().bit_is_set() {
-            return Err(Error::ArbitrationLost);
+            cfg_if::cfg_if! {
```

Comment:
Mind adding a comment explaining why this needs to be checked?
The code looks a bit bizarre out of context.

Reply:
Ah